test/benchmarks/mdc/menu/menu.perf-spec.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {$, by, element, ElementFinder, Key} from 'protractor';
+import {runBenchmark} from '@angular/dev-infra-private/benchmark/driver-utilities';
+
+// Clicking to close a menu is problematic. This is a solution that uses `.sendKeys()` avoids
+// issues with `.click()`.
+
+async function closeMenu(trigger: ElementFinder) {
+  const backdropId = await trigger.getAttribute('aria-controls');
+  if (await $(`#${backdropId}`).isPresent()) {
+    await $(`#${backdropId}`).sendKeys(Key.ESCAPE);
+  }
+}
+
+describe('menu performance benchmarks', () => {
+  it('opens a basic menu', async () => {
+    let trigger: ElementFinder;
+    await runBenchmark({
+      id: 'basic-menu-open',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async () => {
+        trigger = element(by.buttonText('Basic Menu'));
+      },
```

Comment:
For the future: We should remove the `setup` property from the `runBenchmark` function and just do the setup before. e.g. you could just directly assign the `trigger` variable based on what I can tell.

Reply:
You mean inside the `work` callback?

Reply:
No, I just meant before calling `runBenchmark`.

Reply:
The browser is not open until `runBenchmark` is called.

Reply:
IIRC the browser is always open with protractor, it just is on an empty page on initialization. So ideally IMO, the test would manually call `await browser.open(<..>)` before and then do all preparation work before starting with the samples.

The reason why I'm proposing this is that `setup` and `prepare` are very unclear and ambiguous IMO, and as it looks, it's also complicating the logic for preparations (as seen with above for finding the trigger). I don't feel very strongly about it. Changing the name of these properties might help too.

Reply:
Changing `runBenchmark` to not call `openBrowser` would require fixing all the old tests in angular/angular, too. Same with renaming `setup`, `prepare`, and `work`.

Reply:
Yeah, that's clear to me. Though I think that these usages should ideally not prevent us from maintaining good health and readable code. I mentioned this here as a side note, but I think we should do _something_ in the future.

For the record: I'd be happy to go through the instances in the future if needed. We could also add a separate helper that has the less ambiguous API. All of that is obviously not a priority right now but something to keep in mind since we keep adding new benchmarks and spread this across repositories.

Reply:
cc. @josephperrott